### ISSUE TITLE
Time to complete empty tests increases quadratically (~O(n^2)) as the number of tests grows

### ISSUE THREAD

**Issue by cherouvim** [bats-core/bats-core#1036]
I am on a fresh Ubuntu 24 VM and have installed `Bats 1.11.1` using:
```bash
git clone https://github.com/bats-core/bats-core.git /tmp/bats
sudo /tmp/bats/install.sh /usr/local
```

I have empty tests of this form with unique titles so bats will not complain:
```bash
#!/usr/bin/env bats

@test "test 00481268-6f00-460e-b5e9-ed31520f5869" { }
@test "test 00a062d7-473a-4549-988d-ac7585110c89" { }
@test "test 0168e5d7-231b-4a19-b652-364948971c7e" { }
@test "test 027d81cb-7e48-42d7-b7bb-9d8d0ebab2da" { }
@test "test 029ef1b9-fe4f-4081-8780-5552d1a5a662" { }
...
```

Running the above with just 1 test is fast. Doing so with 10 is fast as well. Doing so with 100 or 500 is slow, and doing so with 1000 takes ages. And not only does each test statement take longer, but script initialization (even before the first test runs) is very slow on 500 or 1000 tests. So, the total time to complete all tests does not seem to be linear:

1 test:
```
...
1 test, 0 failures


real    0m0.241s
user    0m0.199s
sys     0m0.050s
```

10 tests:
```
...
10 tests, 0 failures


real    0m0.719s
user    0m0.426s
sys     0m0.298s
```

100 tests:
```
...
100 tests, 0 failures


real    0m7.413s
user    0m5.834s
sys     0m1.531s
```

500 tests:
```
...
500 tests, 0 failures


real    1m12.522s
user    0m53.700s
sys     0m18.415s
```

1000 tests:
```
...
1000 tests, 0 failures


real    4m2.437s
user    3m4.033s
sys     0m57.090s
```

Thus bats is not usable at all for large tests. For extra context, I want to test around 5000 configured http redirects on a local apache server.

**Comment by martin-schulze-vireso** [comment]
I assume the tests are all in one file? What happens when you split them into multiple files?

**Comment by cherouvim** [comment]
@martin-schulze-vireso yes, I am talking about 100, 500, and 1000 tests in a single file. Splitting them to multiple files solves the issue.

**Comment by martin-schulze-vireso** [comment]
Without having looked into the code, my guess would be the duplicate test check. We can probably improve it via sort and uniq.

**Comment by martin-schulze-vireso** [comment]
After looking into it for a bit I can confirm the the duplicate code check will run in `O(n^2)`, since each test is checked against all previous tests.
